### DISCUSSION TITLE
Please mention Andrey Afletdinov more explicitly on the main page

### DISCUSSION THREAD

**Discussion by gustafbstrom** [ihhub/fheroes2#4186] [General]
Hi,

I just want to give credits where credit is due. Andrey Afletdinov, the original developer of this code base, wrote like 99% (estimated guess) of the original code base (found on Sourceforge) before it was put here on Github. I think he should be much more clearly shown as the original developer of this game engine on the main page than simply "Initially, the project was developed on sourceforge."

P.S. Out of curiosity, does anyone know what happened to him? He stopped developing the project years ago, I cannot find any sign of him anywhere on Github from traces left on his sourceforge user page.

**Comment by ihhub** [comment]
Hi @gustafbstrom ,

We mention the main member of the previous team in our credits page and we put a link to Sourceforge on the main page as well. We respect the work of the previous developers. However, the statement of `99% of code written by Andrey Afletdinov` is incorrect for the previous and especially for the current project. Over time we rewrote about 90% of the original code without noting of adding of a new content. From user perspective the game might look the same but from the development point of view is a colossal difference. Speaking of the core engine which is the rendering engine itself, it was fully rewritten by our team as well. Most of parts of the code were replaced or removed like AI, map editor, modding ability and etc. Also it's worth to mention the enormous work of our QA and support team.

I'm not trying to belittle the work of the previous team (please don't get me wrong) but frankly speaking the code from Sourceforge was in an absolute mess and we had to fix literally over 1,000 bugs to make it work properly. Fixing a bug usually needs 10 times more efforts than writing a new code because you have to understand the code and fix it properly, sometimes by fully re-writting it. There is no direct measure of work being done but that's a rough explanation of it. Based on the state of the project in Sourceforge I'd say it was in a very early alpha stage, something like 0.2 or 0.3.

I hope this answers your concerns about the credits page :)

Answering your last question: none of members of the previous team participate in the current development and we have no contacts with them.

**Reply by gustafbstrom** [reply]
Hi @ihhub!
Thanks for taking your time to give a well formulated and explanatory reply – I buy it.

I can clearly see that the current state of the project has a good quality in terms of the compiled result and that the project is very much active. I appreciate all efforts from anyone involved!

Thanks again, and good luck!

**Reply by ihhub** [reply]
Hi @gustafbstrom , you're very welcome! If you have any more questions or suggestions please feel free to tell us🙂
